corehq/apps/accounting/migrations/0112_custom_icon_badges_priv.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 5.2.6 on 2025-11-03 19:21
+
+from django.core.management import call_command
+from django.db import migrations
+
+from corehq.apps.accounting.models import SoftwarePlanEdition
+from corehq.privileges import CUSTOM_ICON_BADGES
+
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+@skip_on_fresh_install
+def _grandfather_privilege(apps, schema_editor):
+    call_command('cchq_prbac_bootstrap')
+
+    skip_editions = ','.join((
+        SoftwarePlanEdition.PAUSED,
+        SoftwarePlanEdition.FREE,
+        SoftwarePlanEdition.STANDARD,
+        SoftwarePlanEdition.PRO,
+    ))
+    call_command(
+        'cchq_prbac_grandfather_privs',
+        CUSTOM_ICON_BADGES,
+        skip_edition=skip_editions,
+        noinput=True,
+    )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('accounting', '0111_rename_is_auto_invoiceable_billingaccount_require_auto_pay'),
+    ]
+
+    operations = [
+        migrations.RunPython(
+            _grandfather_privilege,
+            reverse_code=migrations.RunPython.noop,
```

Comment:
I think this migration is okay, but I have found that I sometimes need to roll back migrations that add privileges, because of conflicts and/or removing a branch from staging.

For that I think you need a function like this:
```python
def _revoke_privilege(apps, schema_editor):
    from corehq.apps.hqadmin.management.commands import cchq_prbac_bootstrap

    call_command(
        'cchq_prbac_revoke_privs',
        CUSTOM_ICON_BADGES,
        skip_edition=skip_editions,
        delete_privs=False,
        check_privs_exist=True,
        noinput=True,
    )
    command = cchq_prbac_bootstrap.Command()
    command.OLD_PRIVILEGES.append(CUSTOM_ICON_BADGES)
    call_command(command)
```

Reply:
I did see that in the recent migrations for GA process and I chose to skip adding that for now.
For Staging, I will run that manually just in case needed.